### PR TITLE
ENT-3489: Save Nudge Emails Info in OfferAssignmentEmailSentRecord

### PR DESCRIPTION
**Pull request by sameenfatima78**
As the SRE team now has dropped the newly added columns in stage db which got added due to a migration that got partially applied, we are finally good to merge the PRs again. This will undo the revert of ENT-3489 #3279

### CODE REVIEW

**Review by saleem-latif — APPROVED** [review]

File: ecommerce/extensions/offer/models.py [hunk 1]
```diff
@@ -609,6 +612,9 @@ class Meta:
 
 
 class OfferAssignmentEmailTemplates(AbstractBaseEmailTemplate):
+    """
+    This model keeps track of all the Assign/Remind/Revoke template saved via frontend portal.
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/offer/models.py]: nit: "This model keeps track of the Assign/Remind/Revoke templates saved via frontend portal."

File: ecommerce/extensions/offer/models.py [hunk 2]
```diff
@@ -686,6 +712,9 @@ def create_record(cls, offer, meta_data=None):
 
 
 class CodeAssignmentNudgeEmailTemplates(AbstractBaseEmailTemplate):
+    """
+    This models keeps track of all the saved templates for nudge emails.
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/offer/models.py]: nit: models -> model

File: ecommerce/extensions/offer/models.py [hunk 3]
```diff
@@ -736,6 +765,11 @@ def get_email_content(self, user_email, code):
 
 
 class CodeAssignmentNudgeEmails(TimeStampedModel):
+    """
+    This model keeps track of all the nudge emails that are to be sent on a specific date based on the info if the
+    user is subscribed to receive emails or not. Once the email is sent, the 'already_sent' flag to set
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/offer/models.py]: nit: "This model keeps track of all the nudge emails that are to be sent on a specific date. This information is based on the user's email subscription preferences."

**Review by irfanuddinahmad — APPROVED** [review]
Some small suggestions ... otherwise looks good!

**Pull request merged**